### DISCUSSION TITLE
Can someone help my dumbass help me with my Java code?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#114446] [Programming Help]
### Body

Since I am a dumbass I am having trouble with Java can you please help me with my dumbass code? 
```java
import javax.swing.*;
public class Window{
    /* Some craptacular code written by someone with an IQ < a dog */
    public static JFrame IAmADodo = new JFrame("Code Editor");
    public static void main(String[] args){
        IAmADodo.setDefaultExit(0);
        IAmADodo.setSize(500, 500); 
    }
}
```

 I don't know what to do because I am for some reason that retarded

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by spenserblack** [comment]
Short answer: It looks like `JFrame.setDefaultExit` doesn't exist. Did you mean to use [`JFrame.setDefaultCloseOperation`](https://docs.oracle.com/javase/8/docs/api/javax/swing/JFrame.html#setDefaultCloseOperation-int-)?

Note that I haven't worked with Java in a *long* time, so I may be incorrect in my assumption of what you want to do.

******

It's pretty obvious from your question and code comments that you're feeling frustrated and/or discouraged. To give some advice:
- Everyone starts as a beginner. When things don't work as intended, try to consider that an opportunity to learn instead of thinking of it as a personal flaw.
- It doesn't really help yourself nor others to self-deprecate. Along with posting your code, try to describe the error(s) you're getting and what you're trying to achieve (i.e. what you are seeing vs what you should be seeing), instead of heaping insults on yourself.
- Projects with GUIs aren't beginner-friendly IMO. If it doesn't *need* to have a graphical interface, consider writing a project as a command-line utility instead, which is generally much easier to write.
- Making a project is a great way to learn a language or tool IMO, but you probably want to start with an easy project. For example, whether I'm learning a language like Ruby or a framework like Electron, my go-to is to make a simple to-do list as a first project.
- Don't feel pressured to know and memorize *everything* about a language's standard library (or any 3rd-party libraries). Focus on the core concepts and ideas, and feel free to go back to the documentation (or do a quick Google search) when you forget exact usage.
- Look for open-source projects relevant to what you want to learn and see if there are any issues that are good for beginners.

**Comment by Mintype** [comment]
As the other comment mentions, `setDefaultExit()` should be replaced with `setDefaultCloseOperation()`. ADDITIONALLY, `setDefaultCloseOperation()` needs a parameter to go in between the parentheses.

Here are some examples:
1. `setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)` exits program on close button pressed.
2. `setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE)` close/exit button does nothing.
3. `setDefaultCloseOperation(JFrame.HIDE_ON_CLOSE)` the window/gui hides/disappears on close button pressed, but the program still runs. 
4. `setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE)` similar to number 3 but the window is de-allocated now.

I hope this helps!
